cranelift/codegen/src/isa/wasm32/lower/isle/generated_code.rs

```diff
@@ -0,0 +1,10 @@
+#![allow(dead_code, unreachable_code, unreachable_patterns)]
+#![allow(unused_imports, unused_variables, non_snake_case, unused_mut)]
+#![allow(
+    irrefutable_let_patterns,
+    unused_assignments,
+    non_camel_case_types,
+    missing_docs
+)] // <-- Why are they allow the bad code?
```

Comment:
```suggestion
)]
```

The isle generated files may contain constructs that trigger these lints as avoiding them would make the emitting code more complex without much benefit.